notify/util.go

```diff
@@ -139,3 +140,57 @@ func GetTemplateData(ctx context.Context, tmpl *template.Template, alerts []*typ
 	}
 	return tmpl.Data(recv, groupLabels, alerts...)
 }
+
+func readAll(r io.Reader) string {
+	if r == nil {
+		return ""
+	}
+	bs, err := ioutil.ReadAll(r)
+	if err != nil {
+		return ""
+	}
+	return string(bs)
+}
+
+// Retrier knows when to retry an HTTP request to a receiver. 2xx status codes
+// are successful, anything else is a failure and only 5xx status codes should
+// be retried.
+type Retrier struct {
+	// Function to return additional information in the error message.
+	CustomDetailsFunc func(code int, body io.Reader) string
+	// Additional HTTP status codes that should be retried.
+	RetryCodes []int
+}
+
+// Process returns a boolean indicating whether the request should be retried
+// and an optional error if the request has failed. If body is not nil, it will
+// be included in the error message.
+func (r *Retrier) Process(statusCode int, body io.Reader) (bool, error) {
```

Comment:
I don't want to bikeshed this, but I feel like `Check` would be a better method name. I guess I don't know. Neither is much butter than the other I suppose.

Reply:
I like `Check` better too.